modules/adspendBidAdapter.js

```diff
@@ -0,0 +1,120 @@
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER, NATIVE, VIDEO } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'adspend';
+const URL = '//rtb.com.ru/headerbidding-bid';
+const COOKIE_NAME = 'hb-adspend-id';
+const UUID_LEN = 36;
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['adspend'],
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+
+  isBidRequestValid: function(bid) {
+    const conf = config.getConfig();
+    return !!(conf.currency && conf.currency.adServerCurrency && bid.crumbs.pubcid);
+  },
+
+  buildRequests: function(bidRequests, bidderRequest) {
+    const req = bidRequests[Math.floor(Math.random() * bidRequests.length)];
```

Comment:
Can you clarify this part further?  It seems like you're just grabbing one request object at random and processing only that.

Reply:
Exactly. Our bidder don't buy more than one ad placement per page.